components/servo/tests/webview.rs

```diff
@@ -611,3 +611,49 @@ fn test_simple_context_menu() {
         ));
     }
 }
+
+#[test]
+fn test_context_menu_reload_action() {
+    let servo_test = ServoTest::new();
+
+    let delegate = Rc::new(WebViewDelegateImpl::default());
+    let webview = WebViewBuilder::new(servo_test.servo())
+        .delegate(delegate.clone())
+        .url(Url::parse("data:text/html,<!DOCTYPE html>").unwrap())
+        .build();
+
+    webview.focus();
+    webview.show(true);
+    webview.move_resize(servo_test.rendering_context.size2d().to_f32().into());
+
+    let load_webview = webview.clone();
+    servo_test.spin(move || load_webview.load_status() != LoadStatus::Complete);
+
+    // Wait for at least one frame after the load completes.
+    delegate.reset();
+    let captured_delegate = delegate.clone();
+    servo_test.spin(move || !captured_delegate.new_frame_ready.get());
+
+    let point = DevicePoint::new(50.0, 50.0).into();
+    webview.notify_input_event(InputEvent::MouseMove(MouseMoveEvent::new(point)));
+    webview.notify_input_event(InputEvent::MouseButton(MouseButtonEvent::new(
+        MouseButtonAction::Down,
+        MouseButton::Right,
+        point,
+    )));
+
+    // The form control should be shown.
+    let captured_delegate = delegate.clone();
+    servo_test.spin(move || captured_delegate.number_of_controls_shown.get() != 1);
+
+    let context_menu = {
+        let mut controls = delegate.controls_shown.borrow_mut();
+        assert_eq!(controls.len(), 1);
+        let Some(EmbedderControl::ContextMenu(context_menu)) = controls.pop() else {
+            unreachable!("Expected embedder control to be a ContextMenu");
+        };
+        context_menu
+    };
+
+    context_menu.select(ContextMenuAction::Reload);
```

Comment:
I think you need to spin here a bit to ensure that the reload is processed. I think you can call `delegate.reset()` and then spin until `delegate.load_status_changed` is true.